core/src/main/scala/kafka/log/stream/s3/metadata/StreamMetadataManager.java

```diff
@@ -222,4 +237,28 @@ public GetObjectsTask() {
         }
     }
 
+    private static class S3StreamImagePair {
```

Comment:
Why do we need S3StreamImagePair? Images themselves are immutable.

Reply:
yes those were immutable. we need atomic replace for the two stream relate metadata Images. 
this package can volatile update and read store in local variable without locks (for multi fields)